app/commands/solution/search_community_solutions.rb

```diff
@@ -73,17 +81,34 @@ def search_query
           must: [
             { term: { 'exercise.id': exercise.id } },
             { term: { status: 'published' } },
-            @criteria.blank? ? nil : { wildcard: { 'user.handle': "*#{criteria}*" } }
+            @sync_status.nil? ? nil : { term: { 'out_of_date': @sync_status.to_sym == :out_of_date } },
+            @mentoring_status.blank? ? nil : { terms: { 'mentoring_status': to_terms(@mentoring_status) } },
+            @tests_status.blank? ? nil : { terms: { 'published_iteration.tests_status': to_terms(@tests_status) } },
+            @head_tests_status.blank? ? nil : { terms: { 'published_iteration.head_tests_status': to_terms(@head_tests_status) } },
+            @criteria.blank? ? nil : {
+              query_string: {
+                query: criteria.split(' ').map { |c| "*#{c}*" }.join(' AND '),
+                fields: ['user.handle', 'published_iteration.code']
```

Comment:
I think the code should probably be a seperate field in the UI. So maybe remove that from here for now?